src/components/LinearSchedule.vue

```diff
@@ -46,52 +59,40 @@ export default {
 	},
 	computed: {
 		sessionBuckets () {
-			const buckets = {}
-			for (const session of this.sessions) {
-				const key = session.start.format()
-				if (!buckets[key]) {
-					buckets[key] = []
-				}
-				if (!session.id) {
-					// Remove duplicate breaks, meaning same start, end and text
-					session.break_id = `${session.start}${session.end}${session.title}`
-					if (buckets[key].filter(s => s.break_id === session.break_id).length === 0) buckets[key].push(session)
-				} else {
-					buckets[key].push(session)
-				}
-			}
-
-			return Object.entries(buckets).map(([date, sessions]) => {
-				let sessionBucket = {}
-				switch (this.sortBy) {
-					case 'title':
-						sessionBucket = {
-							date: sessions[0].start,
-							// sort by room for stable sort across time buckets
-							sessions: sessions.sort((a, b) => {
-								return a.title.localeCompare(b.title)
-							})
-						}
-						break
-					case 'popularity':
-						sessionBucket = {
-							date: sessions[0].start,
-							// sort by room for stable sort across time buckets
-							sessions: sessions.sort((a, b) => {
-								return b.fav_count - a.fav_count
-							})
-						}
-						break
-					default:
-						sessionBucket = {
-							date: sessions[0].start,
-							// sort by room for stable sort across time buckets
-							sessions: sessions.sort((a, b) => this.rooms.findIndex(room => room.id === a.room.id) - this.rooms.findIndex(room => room.id === b.room.id))
-						}
+			if (this.sortBy === 'time') {
+				const buckets = {}
+				for (const session of this.sessions) {
+					const key = session.start.format()
+					if (!buckets[key]) {
+						buckets[key] = []
+					}
+					if (!session.id) {
+						// Remove duplicate breaks, meaning same start, end and text
+						session.break_id = `${session.start}${session.end}${session.title}`
+						if (!buckets[key].some(s => s.break_id === session.break_id)) buckets[key].push(session)
+					} else {
+						buckets[key].push(session)
+					}
 				}
 
-				return sessionBucket
-			})
+				return Object.entries(buckets).map(([date, sessions]) => ({
+					date: sessions[0].start,
+					// sort by room for stable sort across time buckets
+					sessions: _.sortBy(sessions, session => this.rooms.findIndex(room => room.id === session.room.id))
+				}))
+			} else {
+				const sortedSessions = this.sessions.slice().sort((a, b) => {
+					switch (this.sortBy) {
+						case 'title':
+							return a.title.localeCompare(b.title)
+						case 'popularity':
+							return b.fav_count - a.fav_count
+						default:
+							return _.sortBy(this.rooms, room => room.id === a.room.id) - _.sortBy(this.rooms, room => room.id === b.room.id)
```

Comment:
Have you tested this code, how substraction between arrays work?

It seems that `array - array` always returns `NaN`.

Reply:
Hi @hongquan,
you're right, array - array return NaN, 
currently we sort by time, title and popularity, so default case will not be reach, but I have removed it, default will sort by time. Please help me to review it again.